crates/rpc/rpc-eth-api/src/helpers/trace.rs

```diff
@@ -470,4 +449,37 @@ pub trait Trace: LoadState<Evm: ConfigureEvm<Header = Header>> {
     {
         self.trace_block_until_with_inspector(block_id, block, None, insp_setup, f)
     }
+
+    /// Applies chain-specific state transitions required before executing a block.
+    ///
+    /// Note: This should only be called when tracing an entire block vs individual transactions.
+    /// When tracing transaction on top of an already committed block state, those transitions are
+    /// already applied.
+    fn apply_pre_execution_changes<DB: Send + Database<Error: Display> + DatabaseCommit>(
```

Comment:
this makes sense, although this only works as long as the the system calls are the same for eth and op.
but this is a good step